src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -767,126 +769,31 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
       baseContainer.volumeMounts ++ templateVolumeMounts
     )
 
-    val templateNodeSelectors = template.podSpec.map(_.nodeSelector).getOrElse(Map.empty)
+    val templateNodeSelectors = workingTemplate.podSpec.map(_.nodeSelector).getOrElse(Map.empty)
     val mergedNodeSelectors = if (resolvedConfig.nodeSelectors.nonEmpty) {
       resolvedConfig.nodeSelectors
     } else {
       templateNodeSelectors
     }
 
-    val finalPodSpec = template.podSpec
+    val finalPodSpec = workingTemplate.podSpec
       .getOrElse(PodSpec())
       .withTerminationGracePeriodSeconds(0)
       .withRestartPolicy("Never")
       .withInitContainers(Seq(finalInitContainer))
       .withContainers(Seq(finalContainer))
       .withVolumes(mergedVolumes)
       .withNodeSelector(mergedNodeSelectors)
+      .withSecurityContext(new PodSecurityContext().withRunAsUser(resolvedConfig.runAsUser))
```

Comment:
I think it is better to do a merge on the PodSecurityContext, and override the `runAsUser` with either CLI value or default runAsUser value.